cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -136,59 +138,73 @@ private MacroMetadata getMacroFunctionMetadata(int startIndex, int endIndex, Str
 
   private int getStartIndex(String str) {
     int startIndex = str.lastIndexOf("${");
-    while (escaped(startIndex, str)) {
-      // TODO: Break if doubly-escaped syntax
+    while (isEscaped(startIndex, str)) {
       startIndex = str.substring(0, startIndex - 1).lastIndexOf("${");
     }
     return startIndex;
   }
 
   private int getEndIndex(String str, int startIndex) {
     int endIndex = str.indexOf('}', startIndex);
-    while (escaped(endIndex, str)) {
-      // TODO: Break if doubly-escaped syntax
+    while (isEscaped(endIndex, str)) {
       endIndex = str.indexOf('}', endIndex + 1);
     }
     return endIndex;
   }
 
   private int getArgsStartIndex(String str, String macroStr) {
     int argsStartIndex = macroStr.indexOf('(');
-    while (escaped(argsStartIndex, str)) {
-      // TODO: Break if doubly-escaped syntax
-      argsStartIndex = macroStr.indexOf('(', argsStartIndex);
+    while (isEscaped(argsStartIndex, str)) {
```

Comment:
little confused about why this takes 2 strings as arguments. What is the first str and why do we need to use it instead of macroStr?